fuzz/wrap.c

```diff
@@ -100,17 +105,27 @@ extern uid_t __wrap_geteuid(void) {
 extern int __real_open(const char *pathname, int flags);
 extern int __wrap_open(const char *pathname, int flags);
 extern int __wrap_open(const char *pathname, int flags) {
+
   if (prng_up && uniform_random(400) < 1)
     return -1;
+
   /* open write-only files as /dev/null */
   if ((flags & O_ACCMODE) == O_WRONLY)
     return __real_open("/dev/null", flags);
+
   /* FIXME: special handling for /dev/random */
   if (strcmp(pathname, "/dev/urandom") == 0)
     return __real_open(pathname, flags);
+
+  if (conf_file_path && strcmp(pathname, conf_file_path) == 0) {
```

Comment:
Should this move after the O_RDONLY check so that we can assert that for config files also?

```c
assert((flags & O_ACCMODE) == O_RDONLY);

if (conf_file_path && strcmp(pathname, conf_file_path) == 0) {
    assert(*pathname == '/'); /* should not load config from relative path */
    return dup(conf_file_fd);
}

return dup(authfile_fd);
```